src/index.ts

```diff
@@ -85,8 +85,7 @@ function makeBlobServiceClient(config: Config) {
             return new BlobServiceClient(serviceBaseURL, new DefaultAzureCredential());
         }
         default: {
-            const exhaustiveCheck: never = config;
-            throw new Error(exhaustiveCheck);
+            throw new Error('config.authType value is not supported');
```

Comment:
@jakeFeldman Would you be open to something like zod for IO type validation? https://www.npmjs.com/package/zod

Reply:
I'd prefer to keep the bundle size as small as possible. Looks like [zod](https://bundlephobia.com/package/zod@3.22.4) is about 13.2 kb when gzipped. Happy to explore adding this if it will help prevent this issue for users

Reply:
@jakeFeldman Zod would just help to declare a schema for incoming variables and would be able to emit some useful errors.

If there is a way to gain more type safety on incoming variables in the way the plugin is provided to strapi, that would be better, but out of my expertise.

Reply:
IMHO, merging this PR would be useful and could hurt noone.
Today, we've encountered this shady error after upgrading from `3.1.3` to `3.4.0`.
It would be nice to see this human-readable error message <3

Reply:
@TamasHodi It would harm the ability of the code to compile. 

The author just needs to retain the `never` and enrich with a message.

Reply:
@TamasHodi yes the package could handle this better. I'd be okay moving to zod if someone wants to take on the changes. If not, happy to update just to using the `never` type as well. Sorry I don't have the bandwidth to take this on right now but will review and publish any changes. Thanks again